commands/release/release.go

```diff
@@ -91,6 +90,70 @@ var ReleaseCmd = &cobra.Command{
 
 var emptyDomain []string = nil
 
+func parseChangeLogFile(changelogFile []byte) (bool, error) {
+	isBreakingChange := false
+	tbdHeaderFoundCount := 0
+	firstNonEmptyRowNumber := 0
+	lastVersionReleaseRowNumber := 0
+
+	numOfRow := 1
+	scanner := bufio.NewScanner(bytes.NewReader(changelogFile))
+
+	for scanner.Scan() {
+		if versionToBeReleasedPlaceholderHeaderRegex.Match(scanner.Bytes()) {
+			tbdHeaderFoundCount = 1
+			break
+		}
+		numOfRow = numOfRow + 1
+	}
+
+	// No TBD header was found
+	if tbdHeaderFoundCount != expectedNumTBDHeaderLines {
+		return isBreakingChange, stacktrace.NewError("TBD header not found while reading changelog.md")
+	}
+
+	for scanner.Scan() {
+		if versionToBeReleasedPlaceholderHeaderRegex.Match(scanner.Bytes()) {
+			return isBreakingChange, stacktrace.NewError("Found more than %d TBD headers, there can only be #d TBD header in the changelog", expectedNumTBDHeaderLines)
+		}
+
+		// Scan file until next version header detected, searching for first not empty line along the way
+		if lastVersionReleaseRowNumber > 0 && firstNonEmptyRowNumber > 0 {
```

Comment:
@mieubrisse updated the logic, and i have the test for this usecase with the current approach. The reason I did not stick to boolean as suggested, because I only scan the file once to check for out of place tbds as well. That's why i need to keep track of the row numbers.. if the first non empty line (other than TBD) and version header is the same, it means that there's no new changes.